### DISCUSSION TITLE
Feature suggestion - Add button on Manger menu for installing workflow collections

### DISCUSSION THREAD

**Discussion by Suzie1** [Comfy-Org/ComfyUI-Manager#16] [Ideas]
There are a number of creators that create workflow collections to share with other ComfyUI users. It would be nice if these could be installed and updated via ComfyUI manager.

**Comment by ltdrdata** [comment]
Actually, providing workflow downloads was one of the original plans. However, since workflows are intended to be loaded and used on the client side, it raises the question of how to appropriately deliver them.

**Comment by Suzie1** [comment]
RockOfFire suggested if a \workflow folder exists in the \ComfyUI folder then to install it there.

If there is no existing folder then maybe create one or display a message suggesting the user creates one.

**Comment by wutipong** [comment]
Looks like I'm late in the game. Just thought about this kind of feature and then looking into more information, then I found this.

I've checked the code, both the "Manager" custom node and some of the Comfy's frontend code. Looks like the workflow loading is on the frontend side (web page). It wouldn't be possible to access resource on the server/backend side without API to load it. 

I'm not super familiar with how the custom nodes work and what limitations it has. I can't even find how the "Manager" button is added on the toolbar of Comfy UI yet. I think, if we can implement API to list all the workflow files on the server, and the one that returns the content of the input filename, then it would be possible to implement this feature I think. Otherwise, it would be on the Comfy's side to implement this.

**Reply by ltdrdata** [reply]
Actually, I add endpoints like this to fetch information from the backend.
https://github.com/ltdrdata/ComfyUI-Manager/blob/3b57f74450b51bca8bb72cf7b342bcff784685a0/__init__.py#L531

Fetching and displaying a list of workflows from a specified location itself is not an issue. The problem lies in determining how to manage them. Custom nodes are typically managed in the form of git repositories, but workflows, by their nature, are not managed in such a way except by some creators.

**Reply by wutipong** [reply]
> Fetching and displaying a list of workflows from a specified location itself is not an issue. The problem lies in determining how to manage them. Custom nodes are typically managed in the form of git repositories, but workflows, by their nature, are not managed in such a way except by some creators.

Personally I'd go with using git repositories, just like custom nodes. Many creators don't do it that way, yet, but I think once they see it's support people would create repositories to keep the workflow files (even if they just collect them from somewhere else).

But, since people can create workflow file easier than custom nodes, it would be nice if we can have a feature to manually add repository into the list of repositories to pull workflow files from.

**Comment by ppbrown** [comment]
Seconded, for "need a nice GUI workflow loader".
I was hoping that "ComfyUI-Manager" was going to be that tool. I was actually disapointed to find out today, that it only manages custom nodes.
it's painful to have to name workflow files with really long names, in an attempt to remind yourelf what they do... then come back a few days later and STILL not remmeber what they actually do. Becuase filenames are too short for that kind of info.
There needs to be a way to show more detailed info when loading workflows, in ComfyUI.

**Reply by ltdrdata** [reply]
The uploading feature of the workflow will soon be added to the Manager. However, the sharing functionality will be handled on https://comfyworkflows.com/. It's a good idea to suggest improvements for these changes at https://discord.gg/sm4UKkT3.

**Comment by wutipong** [comment]
I'm currently using this for workflow load/save:  https://github.com/pythongosssss/ComfyUI-Custom-Scripts 

However I'm willing too wait for the manager, just so you know :).
